pkg/dao/package_groups.go

```diff
@@ -0,0 +1,405 @@
+				totalPackageGroups,
+				DBErrorToApi(err)
+		}
+		return api.RepositoryPackageGroupCollectionResponse{},
+			totalPackageGroups,
+			&ce.DaoError{
+				NotFound: true,
+				Message:  "Could not find repository with UUID " + repositoryConfigUUID,
+			}
+	}
+
+	repositoryConfig := models.RepositoryConfiguration{}
+	// Select Repository from RepositoryConfig
+
+	if err := r.db.
+		Preload("Repository").
+		Find(&repositoryConfig, "uuid = ?", repositoryConfigUUID).
+		Error; err != nil {
+		return api.RepositoryPackageGroupCollectionResponse{}, totalPackageGroups, err
+	}
+
+	filteredDB := r.db.Model(&repoPackageGroups).Joins(strings.Join([]string{"inner join", models.TableNamePackageGroupsRepositories, "on uuid = package_group_uuid"}, " ")).
+		Where("repository_uuid = ?", repositoryConfig.Repository.UUID)
+
+	if search != "" {
+		containsSearch := "%" + search + "%"
+		filteredDB = filteredDB.
+			Where("name LIKE ?", containsSearch)
+	}
+
+	sortMap := map[string]string{
+		"id":          "id",
+		"name":        "name",
+		"description": "description",
+		"packagelist": "packagelist",
+	}
+
+	order := convertSortByToSQL(sortBy, sortMap, "name asc")
+
+	filteredDB = filteredDB.
+		Order(order).
+		Count(&totalPackageGroups).
+		Offset(offset).
+		Limit(limit).
+		Find(&repoPackageGroups)
+
+	if filteredDB.Error != nil {
+		return api.RepositoryPackageGroupCollectionResponse{}, totalPackageGroups, filteredDB.Error
+	}
+
+	// Return the package group list
+	repoPackageGroupResponse := r.RepositoryPackageGroupListFromModelToResponse(repoPackageGroups)
+	return api.RepositoryPackageGroupCollectionResponse{
+		Data: repoPackageGroupResponse,
+		Meta: api.ResponseMetadata{
+			Count:  totalPackageGroups,
+			Offset: offset,
+			Limit:  limit,
+		},
+	}, totalPackageGroups, nil
+}
+
+func (r packageGroupDaoImpl) RepositoryPackageGroupListFromModelToResponse(repoPackageGroup []models.PackageGroup) []api.RepositoryPackageGroup {
+	repos := make([]api.RepositoryPackageGroup, len(repoPackageGroup))
+	for i := 0; i < len(repoPackageGroup); i++ {
+		r.modelToApiFields(&repoPackageGroup[i], &repos[i])
+	}
+	return repos
+}
+
+// apiFieldsToModel transform from database model to API request.
+// in the source models.PackageGroup structure.
+// out the output api.RepositoryPackageGroup structure.
+//
+// NOTE: This encapsulate transformation into packageGroupDaoImpl implementation
+// as the methods are not used outside; if they were used
+// out of this place, decouple into a new struct and make
+// he methods publics.
+func (r packageGroupDaoImpl) modelToApiFields(in *models.PackageGroup, out *api.RepositoryPackageGroup) {
+	if in == nil || out == nil {
+		return
+	}
+	out.UUID = in.Base.UUID
+	out.ID = in.ID
+	out.Name = in.Name
+	out.Description = in.Description
+	out.PackageList = in.PackageList
+}
+
+func (r packageGroupDaoImpl) Search(orgID string, request api.SearchPackageGroupRequest) ([]api.SearchPackageGroupResponse, error) {
```

Comment:
For some reason when I search EPEL 9 or EPEL 7, I get `[]`. It looks like it introspected correctly and I see its package groups in the database, but I don't see anything in the search.

The search for EPEL 8 works just fine.

```
POST http://localhost:8000/api/content-sources/v1.0/package_groups/names
x-Rh-Identity: eyJpZGVudGl0eSI6eyJ0eXBlIjoiVXNlciIsInVzZXIiOnsidXNlcm5hbWUiOiJzbmFwVXNlciJ9LCJhY2NvdW50X251bWJlciI6IjEwMDgiLCJpbnRlcm5hbCI6eyJvcmdfaWQiOiIxMDA4In19fQo=
Content-Type: application/json

{
  "urls": ["https://dl.fedoraproject.org/pub/epel/9/Everything/x86_64/"],
  "search": "xfce"
}
```

Reply:
Do you see that too? If so, there might be something up with the search query. Trying to debug.

Reply:
that's weird. i tried the search api with these payloads and receive the correct data:

`{
  "search": "xfce",
  "urls": [
    "https://dl.fedoraproject.org/pub/epel/9/Everything/x86_64/"
  ]
}`

`{
  "search": "xfce",
  "urls": [
    "https://dl.fedoraproject.org/pub/epel/8/Everything/x86_64/"
  ]
}`

`{
  "search": "xfce",
  "urls": [
    "https://dl.fedoraproject.org/pub/epel/7/x86_64/"
  ]
}`

which repo urls and what did your request look like? i see that i get `[]` if introspection failed though

Reply:
Looks like I was just using the wrong org ID. No issue here!